src/Sylius/Bundle/ResourceBundle/Controller/ResourcesResolver.php

```diff
@@ -30,11 +30,16 @@ public function getResources(RequestConfiguration $requestConfiguration, Reposit
             return $resources;
         }
 
-        if (!$requestConfiguration->isPaginated() && !$requestConfiguration->isLimited()) {
+        if (
+            ! $requestConfiguration->isPaginated() &&
```

Comment:
We should revert a conditions, here and use `OR` instead of `NOT AND`. It will reduce number of require comparisons and reduce complicity. 

Another thing is some extraction. Maybe some private method? But I have problem with naming for it.